application/browser/application_system.cc

```diff
@@ -92,7 +92,9 @@ void ApplicationSystem::CreateExtensions(
     return;  // We might be in browser mode.
 
   extensions->push_back(new ApplicationRuntimeExtension(application));
-  extensions->push_back(new ApplicationWidgetExtension(application));
+  // unregister widget extension if the application type is not widget
```

Comment:
I'd prefer to comment like below or remove the comments,
'Register the widget extension only when the application is widget format.'